### PR TITLE
Adds interactive edit with live preview, code cleanup and refactor fo…

### LINKED ISSUE #18: When rendering geoJSON layers, prevent fill style from applying to line objects
As a user, I want to display a variety of shape types on the map. For closes shapes like polygons and circle markers, I want the fill value specified in the record styles to apply to the shape. For non-closed shapes (i.e. lines), I want the fill value to be transparent, since filling a line leads to a non-bordered edge that changes with zoom.

### LINKED ISSUE #27: Use default base layer specified by exhibit
As a Neatline user, I want the default map layer loaded by Leaflet to be that specified by the exhibit's spatial_layer property.

### LINKED ISSUE #28: Make alternate map layers specified by exhibit available through layer switcher
As a Neatline user, I want all layers I have enabled through an exhibit's spatial_layers property to be listed in the layer switcher menu of the Leaflet map view. On selecting a layer in the layer switcher, it should become the visible base layer of the map.

### LINKED ISSUE #29: Apply record style settings to map objects: appearance
As a Neatline user, I want map objects to be styled in their appearance based on the options I have set in their corresponding records' Styles section, including Colors, Opacities, and Dimensions. I also want to see changes I make to these options for an existing record be reflected on the map.

### LINKED ISSUE #45: Prevent double-rendering of map object after drawing tool creation + save
When a new map object is created for a record using the drawing tools, and then the record is saved, the app refreshes its data for the record object from the API response. Since the map object is now included in the record's coverage field, this causes the object to be redrawn on top of the originally drawn version of it that's still on the map. One approach could be to remove any tool-drawn shapes at the point of a successful record save and refresh action.

### PR DESCRIPTION
**Pull request by feralresearch**
Adds interactive edit with live preview
Code cleanup and refactor for components
Adds namespace for CSS
Style and UI adjustments
Support for untitled records
Live edit of spatial layers
Adds Map status alert component
Adds Map color picker component.
Adds visual selection indicator on style form
Fixes initial load issue on style color swatches
Prevents fill style from applying to line objects

Depends on #49 

Closes #27 
Closes #28 
Closes #45 
Closes #29
Closes #18

### CODE REVIEW

File: src/components/ExhibitPublicMap/index.js [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+class ExhibitPublicMap extends Component {
+
+
+	// Event handlers for map editing
+	_onEdited = (e) => {
+		this._onChange();
+	}
+
+	_onCreated = (e) => {
+		// Save geometry when it is created - if we don't have a record ID yet, use -1
```
Comment (akstuhl) [inline src/components/ExhibitPublicMap/index.js:40]: "Save" = persist to hidden form fields in this case, right? Looks like a good approach and legible in the code.

File: src/components/ExhibitPublicMap/index.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+								<EditControl position='topleft' onEdited={this._onEdited} onCreated={this._onCreated} onDeleted={this._onDeleted} onMounted={this._onMounted} onEditStart={this._onEditStart} onEditStop={this._onEditStop} onDeleteStart={this._onDeleteStart} onDeleteStop={this._onDeleteStop} draw={{
+										rectangle: false,
+										marker: false
+									}}/>
+							</FeatureGroup>
+					}
+				</LayersControl>
+
+					{records.map(record => {
+							const isSelected = record === this.props.selectedRecord;
```
Comment (akstuhl) [inline src/components/ExhibitPublicMap/index.js:156]: This section (everything inside the map callback) could be a good one to split out into its own component, but that doesn't need to happen in this PR.

File: src/reducers/initialState-mapPreview.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+			}, {
+				slug: "StamenTerrain",
+				displayName: "Stamen: Terrain",
+				deprecated: false,
+				attribution: 'Map tiles by <a href="http://stamen.com">Stamen Design</a>, under <a href="http://creativecommons.org/licenses/by/3.0">CC BY 3.0</a>. Data by <a href="http://openstreetmap.org">OpenStreetMap</a>, under <a href="http://www.openstreetmap.org/copyright">ODbL</a>.',
+				url: 'http://tile.stamen.com/terrain/{z}/{x}/{y}.png'
+			}, {
+				slug: "GooglePhysical",
+				displayName: "Google: Physical",
+				deprecated: true,
```
Comment (akstuhl) [inline src/reducers/initialState-mapPreview.js:88]: The deprecated flag is a great approach. Might be worth adding a comment above the `deprecated: true` lines in these Google layer types, to highlight that they're deprecated and provide a quick explanation as to why for anyone browsing the code.

**Review by akstuhl — APPROVED** [review]
Looks really good, both in the added record editing functionality and the organization of state management and styling. An inline note about Google layer type deprecation is the only change I'd suggest be made before merging.

File: src/reducers/initialState-mapPreview.js (hunk 3 above)
Comment (feralresearch) [reply src/reducers/initialState-mapPreview.js:88]: Just added comments to the top of the initialstate file for this

File: src/components/ExhibitPublicMap/index.js (hunk 2 above)
Comment (feralresearch) [reply src/components/ExhibitPublicMap/index.js:156]: Agreed, noted for next PR

File: src/components/ExhibitPublicMap/index.js (hunk 1 above)
Comment (feralresearch) [reply src/components/ExhibitPublicMap/index.js:40]: Correct - this is as it worked before (saves to a hidden field, recorded to DB on submit), the preview runs in parallel. The "-1" ID lets us handle the case where an ID doesn't exist yet... there are some issues with this I'm still trying to resolve in terms of when the geometry is drawn, but they are outside the scope of the PR.

File: build/main.css [hunk 4]
```diff
@@ -0,0 +1,149 @@
+/*
```
Comment (jamiefolsom) [inline build/main.css:1]: Comments -- Awesome! 😉

File: public/manifest.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
-  "short_name": "React App",
-  "name": "Create React App Sample",
+  "short_name": "Neatline 3",
+  "name": "Neatline 3 by Performant Software",
```
Comment (jamiefolsom) [inline public/manifest.json]: Nit: might be better to say "Neatline 3, from UVA Scholars' Lab"

**Review by jamiefolsom — APPROVED** [review]
Just one nit, not a blocker 👍

File: public/manifest.json (hunk 5 above)
Comment (feralresearch) [reply public/manifest.json]: Will fix before merge

**Pull request merged**